ra/ra_test.go

```diff
@@ -1336,6 +1358,64 @@ func TestCheckFQDNSetRateLimitOverride(t *testing.T) {
 	test.AssertNotError(t, err, "FQDN set certificate per name exemption not applied correctly")
 }
 
+// TestExactPublicSuffixCertLimit tests the behaviour of issue #2681 with and
+// without the feature flag for the fix enabled.
+// See https://github.com/letsencrypt/boulder/issues/2681
+func TestExactPublicSuffixCertLimit(t *testing.T) {
+	_, _, ra, fc, cleanUp := initAuthorities(t)
+	defer cleanUp()
+
+	// Simple policy that only allows 2 certificates per name.
+	certsPerNamePolicy := ratelimit.RateLimitPolicy{
+		Threshold: 2,
+		Window:    cmd.ConfigDuration{Duration: 23 * time.Hour},
+	}
+
+	// Back the mock SA with counts as if so far we have issued the following
+	// certificates:
+	//   - test.dedyn.io (once)
+	//   - test2.dedyn.io (once)
+	//   - dynv6.net (twice)
+	mockSA := &mockSAWithNameCounts{
+		nameCounts: map[string]int{
+			"dedyn.io":       2,
```

Comment:
Ah, I forgot about the need for this to be on the PSL. I wouldn't worry about it then. But do comment on the test that this has an implicit dependency on the example domain name being in the private public suffix section of the PSL. And maybe include a check that it is?